BusinessLogic/src/main/java/BusinessLogicLayer/IncrementalIDGenerator.java

```diff
@@ -0,0 +1,25 @@
+package BusinessLogicLayer;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Created by HassanMahmud on 09/03/2016.
+ */
+public class IncrementalIDGenerator {
+
+    private AtomicInteger termAtomicInt = new AtomicInteger(0);
+    private AtomicInteger documentAtomicInt = new AtomicInteger(0);
+
+//        UUID randomUUID = UUID.randomUUID();
```

Comment:
UUID is not necessarily thread-safe, in high concurrent contexts - possibly like ours, where we generate many many threads along the app life-cycle. We could potentially do use this if we wrap termIdGenerator and documentPKGenerator as ```synchronized``` methods, to avoid the clashes.